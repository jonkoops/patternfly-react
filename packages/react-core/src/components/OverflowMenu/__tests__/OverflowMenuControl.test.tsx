/**
 * @vitest-environment jsdom
 */
import styles from '@patternfly/react-styles/css/components/OverflowMenu/overflow-menu';
import { render, screen } from '@testing-library/react';
import React from 'react';
import { describe, expect, test } from 'vitest';

import { OverflowMenuContext } from '../OverflowMenuContext';
import { OverflowMenuControl } from '../OverflowMenuControl';

describe('OverflowMenuControl', () => {
  test('basic', () => {
    render(
      <OverflowMenuContext.Provider value={{ isBelowBreakpoint: true }}>
        <OverflowMenuControl data-testid="test-id" />
      </OverflowMenuContext.Provider>
    );
    expect(screen.getByTestId('test-id')).toHaveClass(styles.overflowMenuControl);
  });

  test('Additional Options', () => {
    const { asFragment } = render(<OverflowMenuControl hasAdditionalOptions />);
    expect(asFragment()).toMatchSnapshot();
  });
});
