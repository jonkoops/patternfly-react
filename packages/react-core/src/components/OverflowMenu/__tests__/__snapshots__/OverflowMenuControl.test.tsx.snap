// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`OverflowMenuControl > Additional Options 1`] = `
<DocumentFragment>
  <div
    class="pf-v5-c-overflow-menu__control"
  >
      
  </div>
</DocumentFragment>
`;
